tests/js/tags/admin.js

```diff
@@ -0,0 +1,496 @@
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import ReactTable from 'react-table';
+
+import {shallow} from 'enzyme';
+
+import {Container} from 'widgets/columns';
+
+import {ajax} from 'utils/ajax';
+
+import {TagResourcesButton, TagResourcesLoader} from 'tags/admin/';
+
+import TagResourceManager from 'tags/admin/manager';
+import TagResourceSearch from 'tags/admin/search';
+import TagResourceTable from 'tags/admin/table';
+
+
+test('TagResourcesButton render', () => {
+
+    class MockTagResourcesButton extends TagResourcesButton {
+
+        renderButton () {
+            return 7
+        }
+
+        renderResourceManager () {
+            return 23
+        }
+    }
+    const button = shallow(<MockTagResourcesButton />);
+    expect(button.text()).toBe("723");
+});
+
+
+test('TagResourcesButton renderButton', () => {
+    const button = shallow(<TagResourcesButton />);
+    const evt = {preventDefault: jest.fn()};
+
+    expect(button.state().open).toBe(false)
+    button.instance().handleClick(evt);
+    expect(button.state().open).toBe(true)
+    expect(evt.preventDefault.mock.calls).toEqual([[]])
+    button.instance().handleClick(evt);
+    expect(button.state().open).toBe(false)
+    expect(evt.preventDefault.mock.calls).toEqual([[], []])
+})
+
+
+test('TagResourcesButton renderButton', () => {
+    const button = shallow(<TagResourcesButton />);
+    let result = button.instance().renderButton();
+    expect(result.props.children).toBe("Manage resources for this tag");
+    expect(result.props.onClick).toBe(button.instance().handleClick);
+    button.setState({open: true});
+    result = button.instance().renderButton();
+    expect(result.props.children).toBe("Hide the resource manager for this tag");
+    expect(result.props.onClick).toBe(button.instance().handleClick);
+});
+
+
+test('TagResourcesButton renderResourceManager', () => {
+    const button = shallow(
+        <TagResourcesButton
+           tag='foo'
+           api='bar'
+           project='baz' />);
+    let result = button.instance().renderResourceManager();
+    expect(result).toBe('');
+    button.setState({open: true});
+    result = button.instance().renderResourceManager();
+    expect(result.props.tag).toBe('foo');
+    expect(result.props.api).toBe('bar');
+    expect(result.props.project).toBe('baz');
```

Comment:
This puzzles me. Is this not just testing React? I think we can, and probably should, live without tests like that. They increase the size of the code base and the maintenance burden for very little value.

Reply:
> This puzzles me. Is this not just testing React? 

no, definitely not - its testing that our component is taking its vars from the html element

Reply:
Ah, OK, I missed that there's a level of abstraction here. My bad! :)